gensim/models/wrappers/fasttext.py

```diff
@@ -298,8 +312,24 @@ def load_dict(self, file_handle, encoding='utf8'):
                 char_byte = file_handle.read(1)
             word = word_bytes.decode(encoding)
             count, _ = self.struct_unpack(file_handle, '@qb')
-            assert self.wv.vocab[word].index == i, 'mismatch between gensim word index and fastText word index'
-            self.wv.vocab[word].count = count
+            if bin_only and word not in self.wv.vocab:
```

Comment:
This seems unnecessarily convoluted - can the word be in `self.wv.vocab` if we're loading from bin file only?